File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        if len(prices) <= 1:
            return 0
        minprice = float("inf");
        maxprofit = 0;
        for price in prices:
            minprice = min(minprice,price);
            maxprofit = max(maxprofit, price-minprice);
        return maxprofit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minprice = Integer.MAX_VALUE;
        int maxprofit = 0;
        if (prices.length <= 1) return 0;
        for(int price : prices){
            minprice = Math.min(price,minprice);
            maxprofit = Math.max(price-minprice,maxprofit);
        }
        return maxprofit;
    }
}